src/analytics.ts

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+import { env, extensions, Extension, workspace } from "vscode";
+import * as https from "https";
+import * as querystring from "querystring";
+
+enum EventCategory {
+    Extension,
+	TODOs,
+}
+
+enum EventAction {
+    Activated,
+	Enabled,
+	Disabled
+}
+
+class Analytics {
+	private version: string;
+	private isDevelopment: boolean;
+	constructor() {
+		let packageJson = require('../../package.json');
+        this.version = packageJson.version;
+		this.isDevelopment = process.argv.indexOf("--type=extensionHost") >= 0;
+	}
+
+	logActivation() { this.log(EventCategory.Extension, EventAction.Activated); }
+	logShowTodosToggled(enabled: boolean) { this.log(EventCategory.TODOs, enabled ? EventAction.Enabled : EventAction.Disabled); }
+
+	private log(category: EventCategory, action: EventAction) {
+		let data = {
+			v: "1", // API Version.
+			tid: "UA-2201586-19",
+			cid: env.machineId,
```

Comment:
I don't know if this value will work. When I run this locally I see `'someValue.machineId'. analytics wants this value to be something like a UUID  v4 - https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#clientId.

There's some sample code for creating one here: https://github.com/dart-lang/usage/blob/master/lib/src/uuid.dart.